src/gc.c

```diff
@@ -3040,6 +3231,150 @@ JL_DLLEXPORT jl_value_t *jl_gc_alloc_3w(void)
     return jl_gc_alloc(ptls, sizeof(void*) * 3, NULL);
 }
 
+JL_DLLEXPORT int jl_gc_enable_conservative_gc_support()
+{
+    static jl_mutex_t conservative_gc_lock;
+    static_assert(jl_buff_tag % GC_PAGE_SZ == 0,
+        "jl_buff_tag must be a multiple of GC_PAGE_SZ");
+    if (jl_is_initialized()) {
+        JL_LOCK_NOGC(&conservative_gc_lock);
+        int result = support_conservative_marking;
+        support_conservative_marking = 1;
```

Comment:
`jl_atomic_exchange` and get rid of the lock.

Reply:
Okay, that's easy enough.